src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,55 @@
+package org.opentripplanner.middleware.utils;
+
+import org.opentripplanner.middleware.otp.response.Itinerary;
+import org.opentripplanner.middleware.otp.response.Leg;
+
+import java.util.Collection;
+import java.util.List;
+
+public class ItineraryUtils {
+    private static final Collection<String> OTP_TRANSIT_MODES = List.of("TRAM", "BUS", "SUBWAY", "FERRY", "RAIL", "GONDOLA");
+
+    /**
+     * @return true if at least one of the legs of the specified itinerary is
+     *   a transit leg, and none of the legs is a rental leg (e.g. CAR_RENT, BICYCLE_RENT, etc.).
+     */
+    public static boolean itineraryHasTransitAndNoRentals(Itinerary itinerary) {
+        return itineraryHasTransit(itinerary) && !itineraryHasRental(itinerary);
+    }
+
+    /**
+     * @return true if at least one {@link Leg} of the specified {@link Itinerary} is a transit leg.
+     */
+    public static boolean itineraryHasTransit(Itinerary itinerary) {
+        if (itinerary != null && itinerary.legs != null) {
+            for (Leg leg : itinerary.legs) {
+                if (isTransit(leg.mode)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return true if at least one of the legs of the specified itinerary is a rental leg
+     *   (e.g. CAR_RENT, BICYCLE_RENT, MICROMOBILITY_RENT).
+     */
+    public static boolean itineraryHasRental(Itinerary itinerary) {
+        if (itinerary != null && itinerary.legs != null) {
+            for (Leg leg : itinerary.legs) {
+                if (leg.mode.contains("_RENT")) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
```

Comment:
These 3 methods could be combined into one where each leg is checked for all info. Also, this is missing a check for a ridehail.

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/91/commits/8cc7dd929604fe9ba216dcfe41d5ed59161da22c.